### PR TITLE
Move intro docs from flytesnacks to flyte

### PR DESCRIPTION
**Pull request by ppiegaze**
Covers [DOCS-274](https://unionai.atlassian.net/browse/DOCS-274) which is the first step of [DOCS-273](https://unionai.atlassian.net/browse/DOCS-273)

Move the onboarding guide from flytesnacks to flyte repo without making any other visual changes to the docs site

[DOCS-274]: https://unionai.atlassian.net/browse/DOCS-274?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCS-273]: https://unionai.atlassian.net/browse/DOCS-273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/introduction/core_use_cases/analytics.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
```
Comment (cosmicBboy) [inline docs/introduction/core_use_cases/analytics.md:1]: we can get rid of these yaml frontmatter sections, it's added [here](https://github.com/flyteorg/flyte/blob/master/docs/_ext/auto_examples.py#L69-L80) automatically at build time.

File: docs/introduction/core_use_cases/data_engineering.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
+---
```
Comment (cosmicBboy) [inline docs/introduction/core_use_cases/data_engineering.md:1]: remove yaml frontmatter

File: docs/introduction/core_use_cases/index.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (cosmicBboy) [inline docs/introduction/core_use_cases/index.md:1]: remove yaml front matter

**Comment by cosmicBboy** [comment]
I updated `conf.py` to always use jupytext to convert `.md` files to notebooks. I also added back the yaml frontmatter, but just the minimal amount of config needed to render code. This will only be needed in `.md` files that contain `{code-cell} ipython3` blocks, since the myst extension needs to know how to execute the code in these cells.

**Review by cosmicBboy — APPROVED** [review]

**Review by neverett — APPROVED** [review]

**Pull request merged**